system/mp/mpcomm.cpp

```diff
@@ -2237,6 +2237,37 @@ int CMPConnectThread::run()
                 connectHdr.id[0].get(_remoteep);
                 connectHdr.id[1].get(hostep);
 
+                // HPCC-28125 - if remoteep IP != peerEp IP then log warning - do we change remoteep ?
+                //              if hostep != listen ep then log warning - do we change hostep ?
+                // **VERY CONCERNED** about pods and local IPs and Network Address Translation
+
+                if (!_remoteep.ipequals(peerEp))
+                {
+                    StringBuffer rIp, pIp;
+                    _remoteep.getIpText(rIp);
+                    peerEp.getIpText(pIp);
+                    PROGLOG("MP: WARNING: remote client address (%s:%u) != peer address (%s:%u)", rIp.str(), _remoteep.port, pIp.str(), peerEp.port);
+                    _remoteep.set(peerEp);
+                    connectHdr.id[0].set(_remoteep);
+                    // should we sock->close() and continue ?
```

Comment:
Need to test/confirm/try to think of edge cases, but this change looks like the correct fix to me.
It is the peerEp that Dali should be using, not the what the client's notion of what its IP is

Reply:
The concern I have is the port of the remoteep in a valid MP connection is really the other side's server port (how to connect to it), not the ephemeral port in the peerEp and so should we change it here ?